pkg/controller/customresource/customresource.go

```diff
@@ -139,3 +140,34 @@ func SetAnnotation(resource api.AtlasCustomResource, key, value string) {
 	annot[key] = value
 	resource.SetAnnotations(annot)
 }
+
+func ComputeSecretWithFallback[R api.ResourceWithCredentials](fallback bool, project *akov2.AtlasProject, resource R) (*client.ObjectKey, error) {
+	secret, err := resolveConnectionSecret(project, resource)
+	if fallback {
+		return secret, nil
```

Comment:
this logic is very confusing. What are we falling back to? If fallback is `true`, then we never return an error but there might be an error and secret might be `nil`. I don't see the fallback here.

Reply:
yes, the fallback is the global secret. The problem is the fallback logic is elsewhere.

We could move such logic here to be more explicit:
1- Use local credential if present.
2- Otherwise use project credential if present.
3- Otherwise use global credential if passed in.
4- Otherwise fail.

Reply:
In the end I separated away the global fallback logic, and basically left it alone. We can go deeper on a follow up of we would like to make the global fallback optional.

So the current code only choses the local credential if present, fallbacks to the project's one, if present, or returns nil. The function can still error if the input parameters are passed in nil incorrectly.